server/plugins/renderer/renderer.module.js

```diff
@@ -457,7 +457,9 @@ internals.buildReqHeaders = ({
         'stencil-custom-templates': templates,
     };
 
-    if (!request.headers['stencil-config'] && stencilConfig) {
+    const config = request.headers['stencil-config'];
+
+    if ((!config || config === '{}') && stencilConfig) {
```

Comment:
So the header can be undefined and can be empty object?

Reply:
Not sure about undefined, I decided not to touch existing check just not to break anything. However, it definitely could be empty object. I saw that during debugging.